R/simulation.R

```diff
@@ -40,6 +40,18 @@ Simulation <- R6::R6Class(
     #' @field sourceFile Path to the file the simulation was loaded from (read-only)
     sourceFile = function(value) {
       private$.readOnlyProperty("sourceFile", value, private$.sourceFile)
+    },
+    #' @field name Name of the simulation
+    name = function(value) {
+      if (missing(value)) {
+        return(self$get("Name"))
+      } else {
+        # Check for illegal characters
+        if (any(stringr::str_detect(value, paste0("[", getOSPSuiteSetting("illegalCharacters"), "]")))) {
```

Comment:
Why also adding `[` and `]` ?

Reply:
This way it is considered as a regular expression with the symbols inside `[]`. From documentation:

`str_detect(fruit, "[aeiou]")`

will look for the letters `aeiou` inside the string `fruit`.